iceoryx_utils/test/moduletests/test_cxx_newtype.cpp

```diff
@@ -83,8 +83,20 @@ TEST(NewType, CopyAssignableDoesCompile)
 {
     Sut<cxx::NewType<int, newtype::ConstructByValueCopy, newtype::CopyAssignable, newtype::Comparable>> a(491), b(492),
         c(423);
+
+    // Ignore false positive
+    // See: https://isocpp.org/wiki/faq/strange-inheritance#hiding-rule
+
+#if (__GNUC__ == 9 && (__GNUC_MINOR__ > 1 || (__GNUC_MINOR__ == 1 && __GNUC_PATCHLEVEL__ > 0)))
```

Comment:
Would it be possible to avoid these kind of compiler specific `#if`'s? Instead of disabling the test for certain compilers could we try to rewrite in such a way that it is portable?

Reply:
The issue is that we are getting a false positive -Werror=deprecated-copy warning from this line (due to the copy assignment).
There is no deprecated copy constructor.

Either we need to change the implementation somehow to fix this (I have no idea how), or we need to ignore the warning at this particular line, and apparently this is only possible with GCC > 9.1.
I really don't want to spend time fixing it in this PR... This PR is more of a side-task anyway.

@elBoberido @elfenpiff FYI